File: C++/프로그래머스/0/120871. 저주의 숫자 3/저주의 숫자 3.cpp
#include <string>

using namespace std;

int convert(int start, int end) {
    int count = 0;
    
    for (int i = start + 1; i <= end; i++) {
        // 배수 검사
        if (i % 3 == 0) {
            count++;
            continue;
        }
        
        // 3이 들어있는지 검사
        for (const auto& c: to_string(i)) {
            if(c == '3') {
                count++;
                break;
            }
        }
    }
    
    // 추가된 숫자들도 검사하여 return
    if (count == 0) return end;
    return convert(end, end + count);
}

int solution(int n) {
    return convert(0, n);
}

File: Go/프로그래머스/0/120871. 저주의 숫자 3/저주의 숫자 3.go
func Check3 (n int) bool {
    for ;n > 0; n /= 10 {
        if n % 10 == 3 {
            return true
        }
    }
    return false
}

func solution(n int) int {
    for p := 1; p <= n; p++ {
        if p % 3 == 0 || Check3(p) {
            n++
        }
    }
    return n
}